File: Code/Factory.Java/src/factory/basic/ConcreteFactory.java
package factory.basic;

import factory.entities.ConcreteThing;
import factory.interfaces.IFactory;
import factory.interfaces.IThing;

/**
 * Factory that creates objects of the ConcreteThing class
 */
public class ConcreteFactory implements IFactory {
	
	@Override
	public IThing createThing() {
		  return new ConcreteThing();
		}

}

File: Code/Factory.dotNet/Factory.dotNet/Basic/ConcreteFactory.cs
using Factory.Interfaces;
using Factory.Models;

namespace Factory.Basic
{
    /// <summary>
    /// Factory that creates objects of the ConcreteThing class
    /// </summary>
    public class ConcreteFactory : IFactory
    {
        public IThing CreateThing()
        {
            return new ConcreteThing();
        }
    }
}
